quarkchain/cluster/shard_state.py

```diff
@@ -174,7 +174,8 @@ def get_next_tx(self):
                 mblock_header.hash_prev_root_block
             )
             if (
-                prev_root_header.height
+                prev_root_header is None
```

Comment:
Looks like this is caused by the race condition, but I am still not sure where the race comes?

Reply:
suppose shard 1's root genesis height is 100

1. root block 100 contains a minor block from shard 0, and 
2. that minor block links to an older root block, say root block with height 50
3. then, when shard 1 receives the root block 100 and wants to generate its own minor block with height 1, it will go through root block 100's minor blocks, find out one of them links to root block 50 which doesn't exist in its db, and throw the exception